### ISSUE TITLE
Question: How to get cookies from the QueryResult response?

### ISSUE THREAD

**Issue by aagarwal1012** [zino-app/graphql-flutter#335]
It would be really helpful if any can tell the way to extract 'cookies' from the response {QueryResult}.

**Comment by micimize** [comment]
I think `cookies` are a header, so you'd have to grab them in a link

**Comment by aagarwal1012** [comment]
@micimize, but how can I get headers from the QueryResult response? Since the cookie will be returned only after a successful query request.

**Comment by micimize** [comment]
`QueryResults` don't pass down that kind of raw data - you'll have to check for headers in a custom link and grab cookies when they're set. [Here in HttpLink](https://github.com/zino-app/graphql-flutter/blob/master/packages/graphql/lib/src/link/http/link_http.dart#L94) is where the raw response gets parsed.

A link for handling response cookies automatically would be a good feature, [as apollo seems to](https://www.apollographql.com/docs/react/recipes/authentication/#cookie), but does not currently exist.

**Comment by aagarwal1012** [comment]
@micimize,  #337 adds `headers` parameter for FetchResult and QueryResult, so as to get `set-cookie` value from the response headers. Thanks!

**Comment by micimize** [comment]
closing in favor of considering a `rawFetchResult` getter (#371)
